src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -192,7 +187,10 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                             SyntaxRemoveOptions.KeepDirectives |
                             SyntaxRemoveOptions.AddElasticMarker;
                         var modifiedConstructor = constructor
-                            .RemoveNodes(statementsToRemove, removalOptions)!
+                            .RemoveNodes(statementsToRemove
+                                    // take the parent ExpressionStatementSyntax so we take the semicolon too
+                                    .SelectAsArray(assignment => assignment.GetAncestor<ExpressionStatementSyntax>()),
```

Comment:
this worries me as it basically walks up arbitrary far.  it's also not clear ot me how you know you're in an expressionstatement.  for example, this is legal `=> this.x = x;`